design-documents/testing/static/coding-standard-versioning.md

```diff
@@ -25,12 +25,40 @@ Since we want all users (core developers, community contributors, system integra
         ...
     }
 ```
+## Release cadence
+1.	Critical fixes (error during execution) deployed as ready in isolation.
+2.	Monthly releases that include:
+    - Bug fix to existing rule that prevents false-positive findings.
+    - Implementing unit tests for existing rules.
+    - Removing PHP CodeSniffer OOB rule from Magento Coding Standard `ruleset.xml` file.
+    - Removing Magento rule from Magento Coding Standard (sniff code + `ruleset.xml`).
+    - Adding new `exclude-pattern` to the rule.
+    - Decreasing the `severity`.
+    - Changing `type` from `error` to `warning`.
+    - Adding new OOB rule to Magento Coding Standard `ruleset.xml` file (< severity 10).
+    - Implementing new Magento rule and adding it to `ruleset.xml` file (< severity 10).
+    - Changing the behaviour of existing Magento rule (extending functionality) (< severity 10).
+    - Adding new `include-pattern` to the rule (< severity 10).
+    - Increasing the `severity` (< severity 10).
+3.	Quarterly Magento patch release bundled that include:
+    - Adding new OOB rule to Magento Coding Standard `ruleset.xml` file (>= severity 10).
+    - Implementing new Magento rule and adding it to `ruleset.xml` file (>= severity 10).
+    - Changing the behaviour of existing Magento rule (extending functionality) (>= severity 10).
+    - Adding new `include-pattern` to the rule (>= severity 10).
+    - Changing `type` from `warning` to `error`.
+    - Increasing the `severity` to 10.
+    - Changing platform requirements.
+    - Changing namespace.
+
+Therefore, developers will have an opportunity to be tested against all checks once they are merged and released or on demand by obtaining the latest version of the project from GitHub; however, after patch or minor releases of Magento2 some checks may change in priority or start failing extension submissions.
 
 ## Side Effects
 - No `patch` or `minor` dependency will be possible, so any change may lead to build (code check) failure.
 - If released Magento package will contain `"*"` dependency any update may lead to build (code check) failure.
+- Developers will need to track changes in the project either by paying attention to notifications or by following the project in GitHub, in order to make their "in development" code adherent to the upcoming coding-standard release. This will be especially important for patch releases, and critical for minor releases.
 
 ## Marketplace Notifications
+- A week in advance before each release, a notification containing the upcoming changes will be sent. Additional bugfixes may be included into the release after this date, but all possible effort will be applied to not introducing new checks after the notification had been sent. The notification will inform of the entire scope of release changes: bug fixes, severity/scoring changes, and new checks.
```

Comment:
For quarterly releases with severity 10s', we will need to give more notice, say a month,  to give developers time to make the fixes.  This means at least a month before the quarterly release.  Any issues found during this period (say a new severity 10 rule), will happen for the next release train, which will be another quarter away,.